x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -337,49 +319,55 @@ private static class Status {
 
     private final LongSupplier epochMillisProvider;
 
-    // Since Status is the only field that can be updated, we do not need to synchronize access to
+    // Since xPackLicenseStatus is the only field that can be updated, we do not need to synchronize access to
     // XPackLicenseState. However, if status is read multiple times in a method, it can change in between
     // reads. Methods should use `executeAgainstStatus` and `checkAgainstStatus` to ensure that the status
     // is only read once.
-    private volatile Status status = new Status(OperationMode.TRIAL, true, null);
+    private volatile XPackLicenseStatus xPackLicenseStatus;
+
+    public XPackLicenseState(LongSupplier epochMillisProvider, XPackLicenseStatus xPackLicenseStatus) {
+        this.listeners = new CopyOnWriteArrayList<>();
+        this.usage = new ConcurrentHashMap<>();
+        this.epochMillisProvider = epochMillisProvider;
+        this.xPackLicenseStatus = xPackLicenseStatus;
+    }
 
     public XPackLicenseState(LongSupplier epochMillisProvider) {
         this.listeners = new CopyOnWriteArrayList<>();
         this.usage = new ConcurrentHashMap<>();
         this.epochMillisProvider = epochMillisProvider;
+        this.xPackLicenseStatus = new XPackLicenseStatus(OperationMode.TRIAL, true, null);
```

Comment:
Nit: Can we have these two constructor delegate to the private one?
```suggestion
    public XPackLicenseState(LongSupplier epochMillisProvider, XPackLicenseStatus xPackLicenseStatus) {
        this(new CopyOnWriteArrayList<>(), xPackLicenseStatus, new ConcurrentHashMap<>(), epochMillisProvider);    
    }

    public XPackLicenseState(LongSupplier epochMillisProvider) {
        this(new CopyOnWriteArrayList<>(), new XPackLicenseStatus(OperationMode.TRIAL, true, null), new ConcurrentHashMap<>(), epochMillisProvider);
```

Reply:
done in [`05bdab9` (#94927)](https://github.com/elastic/elasticsearch/pull/94927/commits/05bdab9d211bebf7e7d2c1fb92d16c111fc55dab)